### PR TITLE
Repro #27521 - Query fails when selecting two fields from nested query, when fields are identical except for their join-alias

### PR DESCRIPTION
**Pull request by kamilmielnik**
Repro for #27521

The column names are still incorrect (multiple assertions in this test) and toggling the columns does not work.

image

### CODE REVIEW

File: e2e/test/scenarios/models/reproductions.cy.spec.ts [hunk 1]
```diff
@@ -529,9 +529,13 @@ describe.skip("issue 40635", () => {
   });
 
   function assertVisualizationColumns() {
-    cy.findAllByTestId("header-cell").should("contain", "ID");
-    cy.findAllByTestId("header-cell").should("contain", "Products → ID");
-    cy.findAllByTestId("header-cell").should("contain", "Products_2 → ID");
+    assertTableHeader(0, "ID");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/models/reproductions.cy.spec.ts:532]: Just improving assertions in a similar test that I added earlier

File: e2e/test/scenarios/joins/joins-reproductions.cy.spec.js [hunk 2]
```diff
@@ -1135,3 +1138,105 @@ describe("issue 39448", () => {
...
+      cy.findAllByText("ID").should("have.length", 1);
+      cy.findAllByText("Q1 → ID").should("have.length", 1);
+      cy.findAllByText("Q1 → Orders → ID").should("have.length", 1);
+
+      cy.findByRole("button", { name: "Add or remove columns" }).click();
+      cy.findAllByText("ID").should("have.length", 2);
+      cy.findAllByText("Orders → ID").should("have.length", 1);
+
+      // TODO: add assertions for what happens when toggling all the columns here
+      // See https://github.com/metabase/metabase/issues/27521#issuecomment-1948658757
```
Comment (kamilmielnik) [inline e2e/test/scenarios/joins/joins-reproductions.cy.spec.js:1235]: This does not currently work so it's difficult to write assertions for it

File: e2e/test/scenarios/joins/joins-reproductions.cy.spec.js [hunk 3]
```diff
@@ -1135,3 +1138,105 @@ describe("issue 39448", () => {
...
+
+  it("correctly displays joined question's column names (metabase#27521)", () => {
+    cy.visit("/");
+
+    cy.log("Create Q1");
+    newButton("Question").click();
+    entityPickerModal().within(() => {
+      entityPickerModalTab("Tables").click();
+      cy.findByText("Orders").click();
+    });
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/joins/joins-reproductions.cy.spec.js:1157]: Couldn't we `openOrdersTable()` instead?

Use `mode: notebook` if needed.

File: e2e/test/scenarios/joins/joins-reproductions.cy.spec.js [hunk 4]
```diff
@@ -1135,3 +1138,105 @@ describe("issue 39448", () => {
...
+
+    visualize();
+    assertTableHeader(0, "ID");
+    assertTableHeader(1, "Orders → ID");
+
+    cy.button("Save").click();
+    cy.findByLabelText("Name").clear().type("Q1");
+    modal().button("Save").click();
+    modal().findByText("Not now").click();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/joins/joins-reproductions.cy.spec.js:1187]: Please use `saveQuestion` helper here.

**Comment by nemanjaglumac** [comment]
Did you create the first question `Q1` using the UI instead of API because of the result metadata? Can this issue be reproduced by using the API for the first question instead?

File: e2e/test/scenarios/joins/joins-reproductions.cy.spec.js (hunk 3 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/joins/joins-reproductions.cy.spec.js]: a34e339623a60bb85c1467ff781b0c77a5ac71e6

File: e2e/test/scenarios/joins/joins-reproductions.cy.spec.js (hunk 4 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/joins/joins-reproductions.cy.spec.js]: a9f442b5fa639344799c71ca51960804476e94a1

**Comment by kamilmielnik** [comment]
> Did you create the first question `Q1` using the UI instead of API because of the result metadata?

Yes.

> Can this issue be reproduced by using the API for the first question instead?

Technically yes, but this test asserts a bit more - it also checks whether column names are valid before the question is saved - I wouldn't want to sacrifice these assertions or move them to a separate test.

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**